overlord/configstate/configcore/picfg.go

```diff
@@ -92,13 +93,34 @@ func updatePiConfig(path string, config map[string]string) error {
 	return nil
 }
 
-var (
-	errPiConfigNotSupported = fmt.Errorf("configuring pi-config not supported in current mode")
-)
+type piConfigNotSupportedError struct {
+	reason string
+}
+
+func newPiConfigNotSupportedError(msg string) *piConfigNotSupportedError {
+	return &piConfigNotSupportedError{msg}
+}
+
+func (e *piConfigNotSupportedError) Error() string {
+	return fmt.Sprintf("configuring not supported: %s", e.reason)
```

Comment:
hmm how about `configuration cannot be applied: %s` ?

Reply:
I think I like this better, thank you!